rules/prefer-indexof.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+const eslintTemplateVisitor = require('eslint-template-visitor');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID_FINDINDEX = 'findIndex';
+const messages = {
+	[MESSAGE_ID_FINDINDEX]: 'Use `.indexOf()`, rather than `.findIndex()`, when searching the index of an item.'
+};
+
+const templates = eslintTemplateVisitor();
+
+const objectVariable = templates.variable();
+const argumentsVariable = templates.spreadVariable();
+
+const findIndexCallTemplate = templates.template`${objectVariable}.findIndex(${argumentsVariable})`;
+
+const getSearchedConstant = node => {
```

Comment:
This function can be simplified to use an esquery selector.

```js
// Matches `foo.findIndex(bar => bar === baz)`
ArrowFunctionExpression > BinaryExpression.body[operator="==="]

// Matches `foo.findIndex(bar => {return bar === baz})`
// Matches `foo.findIndex(function (bar) {return bar === baz})`
:matches(ArrowFunctionExpression, FunctionExpression) > BlockStatement[length=0].body > ReturnStatement.body > BinaryExpression.argument[operator="==="]
```

Can you try it?